src/runtime/engines/imgbuild/process.go

```diff
@@ -87,3 +109,47 @@ func (e *EngineOperations) CleanupContainer() error {
 func (e *EngineOperations) PostStartProcess(pid int) error {
 	return nil
 }
+
+func (e *EngineOperations) insertEnvScript() error {
+	if e.EngineConfig.RunSection("environment") && e.EngineConfig.Recipe.ImageData.Environment != "" {
+		sylog.Infof("Adding environment to container")
+		err := ioutil.WriteFile("/.singularity.d/env/90-environment.sh", []byte("#!/bin/sh\n\n"+e.EngineConfig.Recipe.ImageData.Environment+"\n"), 0775)
```

Comment:
@ikaneshiro 
As discussed, don't write out `91-environment.sh`. Just set `SINGULARITY_ENVIRONMENT` and let the usage of that in `%post`, etc... create the file.

Reply:
resolved by commit 5f01168